plugins/modules/events.py

```diff
@@ -0,0 +1,145 @@
+    order:
+        description: Retrieval order for cluster events based on time of events.
+        default: ascending
+        required: false
+        type: string
+    offset:
+        description: Number of events to skip before events retrieval.
+        required: false
+        type: integer
+
+author:
+    - Akash Gopalakrishnan (@agopalak)
+"""
+
+EXAMPLES = r"""
+# Use argument
+- name: List cluster events
+  events
+    action: list
+    cluster_id: "deadmeat-dead-meat-dead-meatdeadmeat"
+    limit: 50
+    order: descending
+    offset: 10
+"""
+
+RETURN = r"""
+cluster_events:
+    description: A list of cluster events
+    type: dict
+    returned: always
+    sample: [
+            {
+                "cluster_id": "46f3094d-8967-4f1d-ad09-d5fdfda3830a",
+                "event_time": "2024-11-08T20:15:44.447Z",
+                "message": "Successfully registered cluster",
+                "name": "cluster_registration_succeeded",
+                "severity": "info"
+            },
+            {
+                "cluster_id": "46f3094d-8967-4f1d-ad09-d5fdfda3830a",
+                "event_time": "2024-11-08T20:15:44.652Z",
+                "infra_env_id": "33f1638b-0419-4bc9-ac20-4baddcf14a30",
+                "message": "Updated image information (Image type is \"minimal-iso\", SSH public key is not set)",
+                "name": "image_info_updated",
+                "severity": "info"
+            },
+            {
+                "cluster_id": "46f3094d-8967-4f1d-ad09-d5fdfda3830a",
+                "event_time": "2024-11-08T20:15:52.436Z",
+                "message": "Updated status of the cluster to pending-for-input",
+                "name": "cluster_status_updated",
+                "severity": "info"
+            }
+        ]
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+API_VERSION = "v2"
+API_URL = f"https://api.openshift.com/api/assisted-install/{API_VERSION}"
+
+# add additional query parameters to the query_params_list
+QUERY_PARAMS_LIST = ["cluster_id","limit", "order", "offset"]
+
+def run_module():
+    module_args = dict(
+        action=dict(type="str", required=False, default="list"),
+        cluster_id=dict(type="str", required=False),
+        # any API query parameters may have to be added here
+        limit=dict(type="int", required=False, default=10),
+        offset=dict(type="int", required=False, default=0),        
+        order=dict(type="str",required=False, default="ascending"),
+    )
+
+    token = os.environ.get('AI_API_TOKEN')
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=False)
+
+    # Set headers
+    headers = {
+        'Content-Type': 'application/json',
+        'Authorization': f'Bearer {token}'
+    }
+
+    # List cluster events
+    if module.params.get('action') == "list":
+        if not module.params.get('cluster_id'):
+            module.fail_json(msg="cluster_id is required for list cluster events action")
```

Comment:
This could be moved into a function and used by multiple files, if we create 2 seperate playbooks.

@tonyskapunk @vjayaramrh